src/checkmatelib/client.py

```diff
@@ -88,12 +88,17 @@ def check_url(  # pylint: disable=inconsistent-return-statements
     def _clean_url(cls, url):
         """Clean the URL before we send it.
 
-        Checkmate will do this for itself, and doesn't trust our input, but
-        this allows us to fail fast for bad URLs, before Checkmate has a chance
-        to check them. We can also apply extra constraints.
+        This applies the same canonicalization to the URL as we do when the
+        real check happens in Checkmate. This means anything which will cause
+        Checkmate to raise `BadURL` will cause us to first, saving some time.
```

Comment:
> "cause us to first"

You mean "cause us to raise here first"?

Reply:
I've cleaned this up to be clearer.